src/cares_wrap.cc

```diff
@@ -1064,30 +1064,28 @@ int ParseSoaReply(Environment* env,
   /* Can't use ares_parse_soa_reply() here which can only parse single record */
   unsigned int ancount = cares_get_16bit(buf + 6);
   unsigned char* ptr = buf + NS_HFIXEDSZ;
-  char* name;
-  char* rr_name;
+  MallocedBuffer<char> name;
+  MallocedBuffer<char> rr_name;
   long temp_len;  // NOLINT(runtime/int)
-  int status = ares_expand_name(ptr, buf, len, &name, &temp_len);
+  int status = ares_expand_name(ptr, buf, len, &name.data, &temp_len);
   if (status != ARES_SUCCESS) {
     /* returns EBADRESP in case of invalid input */
     return status == ARES_EBADNAME ? ARES_EBADRESP : status;
   }
 
   if (ptr + temp_len + NS_QFIXEDSZ > buf + len) {
-    free(name);
     return ARES_EBADRESP;
   }
   ptr += temp_len + NS_QFIXEDSZ;
 
   for (unsigned int i = 0; i < ancount; i++) {
-    status = ares_expand_name(ptr, buf, len, &rr_name, &temp_len);
+    status = ares_expand_name(ptr, buf, len, &rr_name.data, &temp_len);
```

Comment:
[ES.26](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-recycle): Don’t use a variable for two unrelated purposes